src/main/java/com/pinterest/secor/consumer/Consumer.java

```diff
@@ -71,7 +71,7 @@ private void init() throws Exception {
         FileRegistry fileRegistry = new FileRegistry(mConfig);
         UploadManager uploadManager = ReflectionUtil.createUploadManager(mConfig.getUploadManagerClass(), mConfig);
 
-        mUploader = new Uploader(mConfig, mOffsetTracker, fileRegistry, uploadManager);
+        mUploader = ReflectionUtil.createUploader(mConfig.getUploaderClass(), mConfig, mOffsetTracker, fileRegistry, uploadManager);
```

Comment:
If we use reflection to create instance, the standard way is to create the object using no-arg constructor through reflection, and call a configure method with the rest of the arguments.